_posts/2023-02-01-concerns-in-ruby-part-1.md

```diff
@@ -0,0 +1,92 @@
+author_name: Alex Smith
+author_github_username: tradesmanhelix
+excerpt: Pitfalls and best practices for using concerns in Ruby.
+---
+
+In October of 2022, I had the privilege of giving a talk on "Concerns (aka. modules) in Ruby" to the Maxwell development team and afterwards was asked if I could condense my talk into a blog post.
+
+Well, I unfortunately failed at the "condense" part as we'll need several posts to cover everything, but I am excited to present the talk to you in written format and hope that you find it entertaining if not educational.
+
+***
+
+To begin, let's orient ourselves - what are concerns/modules (hereinafter concerns) in Ruby? When you see the word "concern" in this series moving forward, we are talking about the following:
+
+![concerns in Rails](../images/posts/concerns-in-ruby-part-1/img_1.jpg)
+
+Broadly-speaking, **concerns are a mechanism for allowing an otherwise-unrelated set of classes to share code**. In Ruby, they are created using the `module` keyword and in Rails apps they can live in multiple locations but conventionally are found in directories like `app/models/concerns`, `app/controllers/concerns`, etc.
+
+As an example of one place we might use a concern, say we're working on an app for managing a "family fun park" type of business that has a laser tag arena, party rooms, and bowling lanes, and we want to allow our customers to book or reserve any and all of these. We can accomplish this behavior via a `Bookable` concern that adds scheduling functionality to any class that includes it.
+
+![share code via concerns](../images/posts/concerns-in-ruby-part-1/img_2.jpg)
+
+You'll often see concerns referred to as mix-ins because their functionality gets "added to" or "mixed into" the classes that include them. Also note that classes aren't limited to just one concern - they can include as many as they want!
+
+The code sharing made possible by concerns is DRY ("don't repeat yourself") and solves the problem of single inheritance since classes, in Ruby at least, can only have one parent class.
+
+![concerns summary](../images/posts/concerns-in-ruby-part-1/img_3.jpg)
+
+Let's look at some diagrams to help us visualize concerns by comparing them to the more well-known concept of inheritance.
+
+This first diagram depicting classical inheritance should be pretty self-explanatory to anyone familiar with the concept:
+
+![inheritance diagram](../images/posts/concerns-in-ruby-part-1/img_4.jpg)
+
+Now, here's a diagram that depicts how concerns work:
+
+![concerns diagram](../images/posts/concerns-in-ruby-part-1/img_5.jpg)
+
+Hopefully, as you look at the above, a little red flag is waving around annoyingly in your mind. Hopefully, as you examine these diagrams, you are hit with the fact that in both we are looking at _relationships_ between classes, but the relationship created by inheritance is much more explicit/obvious than the relationship that's created when we use a concern.
```

Comment:
@tradesmanhelix , small thing, but you are starting two sentences with "Hopefully". might be a better way to write it.